kafka/consume.go

```diff
@@ -73,33 +81,91 @@ func NewConsumer(log logrus.FieldLogger, conf Config, opts ...ConfigOpt) (*Confl
 		return nil, err
 	}
 
-	confluenceConsumer := &ConfluentConsumer{
+	if conf.Timeout == 0 {
+		conf.Timeout = DefaultTimout
+	}
+
+	cc := &ConfluentConsumer{
 		c:    consumer,
 		conf: conf,
 		log:  log,
 	}
 
-	return confluenceConsumer, nil
+	if cc.conf.Consumer.Partition == nil {
+		// if we are not reading from a specific partition, we subscribe
+		cc.setupRebalanceHandler(cc.conf.Consumer.InitialOffset)
```

Comment:
This should only be called if `cc.conf.Consumer.InitialOffset` is set.

Reply:
The initial offset will be 0 and it will start to read there.  Am I missing something?

Reply:
This is not the default behaviour of a Kafka consumer.  The default behaviour would be reading from the latest stored offset (if exists), otherwise will follow the strategy set in https://github.com/netlify/netlify-commons/blob/1b39178bb1f85791e2a13063b16285456455b673/kafka/consume.go (earliest).

Reply:
I've updated the logic to now consume at the offset provided by the broker if not initial offset is set in the config.